apps/site/src/_pages/docs/4_spec/02_graph-module.mdx

```diff
@@ -1,303 +1,445 @@
 
-**Linked entities**: entities which are linked to the block entity, whether directly or via other entities.
+**Entity type:** an ontology type that defines the structure of an entity and its link entities. An entity type references other entity types and property types.
 
-**Properties:** the fields on a schema, what types of data are permitted as their values, and any other validation rules. Field types may be scalar (integer, string etc.), complex objects, collections (lists/arrays), or refer to another schema.
+**Link Entity type:** an entity type which has a special marker to say it constrains a link entity.
 
-**Schema:** a definition of a data structure, associated with an entity type. Entities of a given type must have properties which conform to the type’s schema.
+**Property type:** an ontology type that defines the structure of an individual property within an entity. A property type references other property types and data types.
+
+**Data type:** an ontology type that defines the possible space of possible values that an individual property can take within an entity.
+
+**Graph:** a network of graph elements (entities, entity types, property types, and data types) connected by edges.
+
+**Edge:** a directed connection between two graph elements which describes their relationship. Possible edges include those that describe references between types, or the relationship between the left and right entities of link entities.
+
+**Subgraph:** a specific subset of a graph, usually returned as a result of a query. Can contain a set of roots which are distinguished elements of the subgraph, and usually correspond to the direct results of the query as the subgraph may contain other elements of the graph that were connected to the root elements through various edges.
+
+**Block entity**: an entity associated with the specific instance of the block, conforming to block's specified entity type.
+
+**Block Entity Subgraph:** a subgraph which contains the block entity as its root, with connected elements to a given depth.
+
+## The Type System
+
+The data model of the graph module is defined through its **Type System**.
+The type system describes how a graph of **entities** is structured, through a composition of the 3 classes of **ontology types**, **data types**, **property types**, and **entity types**.
+
+When composed, these types are capable of describing any JSON object.
+
+### <Hidden>The Type System </Hidden>Data Types
+
+A **data type** describes a space of possible valid _values_.
+These cover all possible primitive values that can be represented within JSON, as well as `Object` and `Null` which are included for completeness of the system ([see the RFC for rationale](https://github.com/blockprotocol/blockprotocol/blob/main/rfcs/text/0352-graph-type-system.md#including-an-empty-list-data-type)).
```

Comment:
`EmptyList`? Your href is specifically rleated to it but it is not mentioned here

Reply:
Oops I wrote `null` instead of `empty list` (`null` is a primitive value after all) 😁 
```suggestion
These cover all possible primitive values that can be represented within JSON, as well as `Object` and `Empty List` which are included for completeness of the system ([see the RFC for rationale](https://github.com/blockprotocol/blockprotocol/blob/main/rfcs/text/0352-graph-type-system.md#including-an-empty-list-data-type)).
```